aws/table_aws_backup_report_plan.go

```diff
@@ -0,0 +1,196 @@
+				Transform:   transform.FromField("ReportPlanName"),
+			},
+			{
+				Name:        "description",
+				Description: "An optional description of the report plan with a maximum 1,024 characters.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ReportPlanDescription"),
+			},
+			{
+				Name:        "creation_time",
+				Description: "The date and time that a report plan is created, in Unix format and Coordinated Universal Time (UTC).",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "deployment_status",
+				Description: "The deployment status of a report plan. The statuses are CREATE_IN_PROGRESS, UPDATE_IN_PROGRESS, DELETE_IN_PROGRESS, and COMPLETED.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "last_attempted_execution_time",
+				Description: "The date and time that a report job associated with this report plan last attempted to run, in Unix format and Coordinated Universal Time (UTC).",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+			{
+				Name:        "last_successful_execution_time",
+				Description: "The date and time that a report job associated with this report plan last successfully ran, in Unix format and Coordinated Universal Time (UTC).",
+				Type:        proto.ColumnType_TIMESTAMP,
+			},
+
+			// JSON Columns
+			{
+				Name:        "report_delivery_channel",
+				Description: "Contains information about where and how to deliver your reports, specifically your Amazon S3 bucket name, S3 key prefix, and the formats of your reports.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "report_setting",
+				Description: "Identifies the report template for the report. Reports are built using a report template.",
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ReportPlanName"),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ReportPlanArn").Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsBackupReportPlans(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create session
+	svc, err := BackupClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_backup_report_plan.listAwsBackupReportPlans", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Limiting the results
+	maxLimit := int32(1000)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
+				maxLimit = 1
+			} else {
+				maxLimit = limit
+			}
```

Comment:
```suggestion
			maxLimit = limit
```

The logic `if limit < 1` is not required. The Steampipe SDK itself has handled it.